src/components/calcite-input/calcite-input.tsx

```diff
@@ -183,27 +240,270 @@ export class CalciteInput {
   }
 
   componentWillLoad(): void {
-    this.defaultValue = this.value;
     this.childElType = this.type === "textarea" ? "textarea" : "input";
-    this.requestedIcon = setRequestedIcon(INPUT_TYPE_ICONS, this.icon, this.type);
-    this.minString = this.min?.toString();
+    this.defaultValue = this.value;
     this.maxString = this.max?.toString();
-    this.stepString = this.step?.toString();
+    this.minString = this.min?.toString();
+    this.requestedIcon = setRequestedIcon(INPUT_TYPE_ICONS, this.icon, this.type);
   }
 
   componentDidLoad(): void {
     this.slottedActionEl = this.el.querySelector("[slot=input-action]");
     this.setDisabledAction();
+    if (this.shouldFormatNumberByLocale()) {
+      this.childEl.style.cssText = hiddenInputStyle;
+    }
   }
 
-  get isTextarea(): boolean {
-    return this.childElType === "textarea";
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * @internal
+   */
+  @Event() calciteInputFocus: EventEmitter;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInputBlur: EventEmitter;
+
+  /**
+   * This event fires when the value of the input changes.
+   */
+  @Event({ eventName: "calciteInputInput", cancelable: true }) calciteInputInput: EventEmitter;
```

Comment:
Just to confirm, the `cancelable` is still relevant to this PR after the internal input refactor, right?

Reply:
Correct!  The cancelability is still relevant even though the color-picker isn't using that custom event directly, its still just relying on the native input event.